lib/ansible/modules/packaging/os/apt_key.py

```diff
@@ -128,6 +146,51 @@ def find_needed_binaries(module):
     module.get_bin_path('grep', required=True)
 
 
+def in_no_proxy(module, server, no_proxy):
+    """ check if given server url is included in the no_proxy list.
+    No DNS lookup is carried out at this stage.
+
+    :param server: domain name, hostname or IP. Port is supported.
+    :param no_proxy: list of comma separated URL. Support domain name, IP, subnet range, wildcard *, trailing and leading .
+    example:
+        no_proxy=example.com, .example.com, 192.168.56., 192.168.*.4, 127.0.0.0/8
+    :return: True if no_proxy contains key_server
+    """
+    urls = no_proxy.split(',')
+    import re
+    # remove trailing ports from key server, example :80
+    k = re.sub(r':\d+$', '', server, re.UNICODE)
+    # remove leading protocal, example: hkp://keyserver.ubuntu.com
+    k = re.sub(r'^\w+://', '', k, re.UNICODE)
+    for url in urls:
+        u = url.strip()
+        if k == u:
+            return True
+        elif u.startswith('.') and k.endswith(u):
+            return True
+        elif u.endswith('.') and k.startswith(u):
```

Comment:
I'd drop this feature, as I'm not aware of any other no_proxy processing that does prefix matching.